algorithms/date_event_extraction/SocketNER.js

```diff
@@ -0,0 +1,113 @@
+var spawn = require("child_process").spawn
+var deasync = require("deasync")
+
+function socketNER(port, classifierFileName, pathToNER) {
+   this.port = port || 1234
+   this.classifier = classifierFileName || "english.muc.7class.distsim.crf.ser.gz"
+   this.pathToNER = pathToNER || "/"
+   this.server = undefined
+   this.client = undefined
+}
+
+socketNER.prototype.startServer = deasync(function (cb) {
+   var self = this
+   self.server = spawn(
+      "java",[
+         "-mx750m", "-cp",
+         self.pathToNER + "stanford-ner.jar",
+         "edu.stanford.nlp.ie.NERServer",
+         "-loadClassifier", self.pathToNER + self.classifier,
+         "-port", self.port, "-outputFormat", "inlineXML"
+      ]
+   )
+   // I don't know why server's stderr stream gets
+   // all output and why stdout don't.
+   self.server.stderr.on("data", reader)
+//  console.log(self.server);
+   // Server would finish loading, when it flushes
+   // out 'done [x secs]'
```

Comment:
I mean the comment about the console/logger error, I noticed now it was not included in the review preview :) If stderr works, lets just use that?